api/swagger.yml

```diff
@@ -1,507 +1,795 @@
+        - creation_date
+        - id
+        - default_branch
+        - storage_namespace
+      properties:
+        id:
+          type: string
+        creation_date:
+          type: integer
+          format: int64
+        default_branch:
+          example: "master"
+          type: string
+        storage_namespace:
+          type: string
+          description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
+
+    RepositoryList:
+      type: object
+      required:
+        - pagination
+        - results
+      properties:
+        pagination:
+          $ref: "#/components/schemas/Pagination"
+        results:
+          type: array
+          items:
+            $ref: "#/components/schemas/Repository"
+
+    MergeResult:
+      type: object
+      required:
+        - summary
+        - reference
+      properties:
+        summary:
+          type: object
+          required:
+            - added
+            - removed
+            - changed
+            - conflict
+          properties:
+            added:
+              type: integer
+            removed:
+              type: integer
+            changed:
+              type: integer
+            conflict:
+              type: integer
+        reference:
+          type: string
 
-  staging_location:
-    type: object
-    description: location for placing an object when staging it
-    properties:
-      physical_address:
-        type: string
-      token:
-        type: string
-        description: opaque staging token to use to link uploaded object
-    required:
-      - token
-
-  staging_metadata:
-    type: object
-    description: information about uploaded object
-    properties:
-      staging:
-        type: object
-        $ref: "#/definitions/staging_location"
-      checksum:
-        type: string
-        description: unique identifier of object content on backing store (typically ETag)
-      mtime:
-        type: integer
-        format: int64
-      size_bytes:
-        type: integer
-        format: int64
-    required:
-      - staging
-      - checksum
-      - mtime
-      - size_bytes
-
-  object_stage_creation:
-    type: object
-    required:
-      - physical_address
-      - checksum
-      - size_bytes
-    properties:
-      physical_address:
-        type: string
-      checksum:
-        type: string
-      size_bytes:
-        type: integer
-        format: int64
-      metadata:
-        type: object
-        additionalProperties:
+    RepositoryCreation:
+      type: object
+      required:
+        - name
+        - storage_namespace
+      properties:
+        name:
+          type: string
+          pattern: '^[a-z0-9][a-z0-9-]{2,62}$'
+        storage_namespace:
+          type: string
+          description: "Filesystem URI to store the underlying data in (e.g. 's3://my-bucket/some/path/')"
+          example: "s3://example-bucket/"
+          pattern: '^(s3|gs|https?|mem|local|transient)://.*$'
```

Comment:
Possibly silly question: Can we tell our code generator to validate this only on the server??  Otherwise we add support for random backing protocol `abc`, and every client has to change.

Reply:
I guess this is an implementation detail - in our case the verification is done on the server side by middleware that validates the swagger spec.